client/json-rpc/src/async_client/client.rs

```diff
@@ -432,39 +488,60 @@ impl<R: RetryStrategy> Client<R> {
             .map(|(i, r)| r.to_json(i))
             .collect();
 
-        let rpc_resps: Vec<JsonRpcResponse> = self.send_json_request(json![&json_requests]).await?;
-        if rpc_resps.len() != requests.len() {
-            return Err(Error::unexpected_invalid_batch_response(rpc_resps));
-        }
-
-        let mut resp_maps: HashMap<usize, JsonRpcResponse> = HashMap::new();
-        for resp in rpc_resps {
-            let id = self.validate(&resp, 0, requests.len() - 1)?;
-            if resp_maps.contains_key(&id) {
-                return Err(Error::unexpected_duplicated_response_id(resp));
-            }
-
-            resp_maps.insert(id, resp);
-        }
-
-        json_requests
-            .iter()
-            .enumerate()
-            .map(|(i, req)| {
-                resp_maps
-                    .remove(&i)
-                    .ok_or_else(|| Error::unexpected_no_response(req.clone()))
+        let rpc_resps_list: Vec<Vec<JsonRpcResponse>> =
+            self.multi_send_json_request(json![&json_requests]).await?;
+        let mut rpc_resps_list = rpc_resps_list
+            .into_iter()
+            .map(|rpc_resps| {
+                if rpc_resps.len() != requests.len() {
+                    return Err(Error::unexpected_invalid_batch_response(rpc_resps));
+                }
+                let mut resp_maps: HashMap<usize, JsonRpcResponse> = HashMap::new();
+                for resp in rpc_resps {
+                    let id = self.validate(&resp, 0, requests.len() - 1)?;
+                    if resp_maps.contains_key(&id) {
+                        return Err(Error::unexpected_duplicated_response_id(resp));
+                    }
+                    resp_maps.insert(id, resp);
+                }
+                json_requests
+                    .iter()
+                    .enumerate()
+                    .map(|(i, req)| {
+                        resp_maps
+                            .remove(&i)
+                            .ok_or_else(|| Error::unexpected_no_response(req.clone()))
+                    })
+                    .collect::<Result<Vec<JsonRpcResponse>, Error>>()
             })
-            .collect()
-    }
-
-    async fn send_json_request<T: for<'de> serde::Deserialize<'de>>(
```

Comment:
here, call the HttpClient to send request

         return self.http_client.send(request)

the default httpclient send method should do same with single_json_request.